src/components/RangeInput/RangeInput.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import autoHideContainerHOC from '../../decorators/autoHideContainer.js';
+import headerFooterHOC from '../../decorators/headerFooter.js';
+
+export class RawRangeInput extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      min: props.values.min,
+      max: props.values.max,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      min: nextProps.values.min,
+      max: nextProps.values.max,
+    });
+  }
+
+  onChange = name => event => {
+    this.setState({
+      [name]: event.currentTarget.value,
+    });
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+
+    this.props.refine([this.state.min, this.state.max]);
+  };
+
+  get isDisabled() {
+    return this.props.min >= this.props.max;
+  }
+
+  render() {
+    const { min: minValue, max: maxValue } = this.state;
+    const { min, max, step, cssClasses, labels } = this.props;
+
+    return (
+      <form className={cssClasses.form} onSubmit={this.onSubmit}>
+        <fieldset className={cssClasses.fieldset}>
```

Comment:
Why is the fieldset necessary?

Reply:
I just try to stay close as much as possible to RIS markup :)